repl/src/main/scala/ammonite/repl/interp/Preprocessor.scala

```diff
@@ -70,7 +92,7 @@ object Preprocessor{
     }
 
     val Expr = Processor{
-      case (name, code, tree) => Output(s"val $name = (\n$code\n)", Seq(pprint(name)))
+      case (name, code, tree) => Output(s"val $name = { () =>\n$code\n}.apply", Seq(pprint(name)))
```

Comment:
Aren't Expr and PatVarDef separate, so only one of them processes a single line of code?